secantTests/Util/KeychainManagerTests.swift

```diff
@@ -0,0 +1,256 @@
+                    XCTAssertEqual(birthdaySet, birthday.intValue, "Keychain: birthday stored and retrieved must be the same.")
+                } else {
+                    XCTFail("Keychain: `testBirthdayStoredSucessfuly` birthday decoding failed.")
+                }
+            } catch {
+                XCTFail("Keychain: `testBirthdayStoredSucessfuly` birthday decoding failed.")
+            }
+        } catch let err {
+            XCTFail("Keychain: no error is expected for `testBirthdayStoredSucessfuly` but received. \(err)")
+        }
+    }
+
+    func testBirthdayDuplicate() throws {
+        do {
+            try storage?.setBirthday(birthdaySet)
+            try storage?.setBirthday(birthdaySet)
+            
+            XCTFail("Keychain: `testBirthdayDuplicate` is expected to throw a `duplicate` error but passed instead.")
+        } catch {
+            guard let error = error as? RecoveryPhraseStorage.KeychainError else {
+                XCTFail("Keychain: the error is expected to be KeychainManager.KeychainManagerError but it's \(error).")
+                
+                return
+            }
+            
+            XCTAssertEqual(error, RecoveryPhraseStorage.KeychainError.duplicate, "Keychain: error must be .duplicate but it's \(error).")
+        }
+    }
+
+    func testBirthdayNoDataFound() throws {
+        do {
+            _ = try storage?.birthday()
+            
+            XCTFail("Keychain: `testBirthdayNoDataFound` should fail but received some data.")
+        } catch {
+            guard let error = error as? RecoveryPhraseStorage.KeychainError else {
+                return XCTFail("Keychain: the error is expected to be KeychainManager.KeychainManagerError but it's \(error).")
+            }
+
+            XCTAssertEqual(error, RecoveryPhraseStorage.KeychainError.noDataFound, "Keychain: error must be .noDataFound")
+        }
+    }
+    
+    func testDeleteBirthday() throws {
+        do {
+            // set the data (do not rely on RecoveryPhraseStorage)
+            do {
+                let data = try NSKeyedArchiver.archivedData(withRootObject: NSNumber(value: birthdaySet), requiringSecureCoding: true)
+                
+                try setData(data, forKey: "zcashBirthday")
+            } catch {
+                XCTFail("`testDeleteBirthday` storing `birthdaySet` failed.")
+            }
+
+            // test deletion of the birthday
+            storage?.deleteBirthday()
+            
+            // check presence of the key/data
+            let data = data(forKey: "zcashBirthday")
+            
+            XCTAssertEqual(data, nil, "Keychain: keychain is expected to not find anything for key `zcashBirthday` but received some data.")
+        }
+    }
+    
+    // MARK: - Misc
+    
+    /// The followings methods are here purposely to not rely on `RecoveryPhraseStorage` in order to test functionality of JUST ONE method at a time
```

Comment:
I've noticed these methods are duplicated from the private functionality that lives inside the app.  The primary logic here is setting up the query though, what would you think about potentially sharing the query data definitions, so that there is less duplication and easier maintenance over time?

Reply:
done